FAQs/FAQs.md

```diff
@@ -0,0 +1,45 @@
+# Using Spark with R - Some common questions & answers
+
+
+## I'm new to Spark. Should I use SparkR or sparklyr?
+
+Spark on R is different to the Python and Scala APIs in that users have to choose between two distinct APIs when writing Spark code. SparkR is the API set that is part of the main Apache Spark project, whilst sparklyr was originated and continues to be maintained by the team behind RStudio.
+
+Your choice of API should be informed by what you are trying to do and you're existing skill set:
+### SparkR
+- Good for data engineering tasks - has the full suite of Spark's data engineering APIs available.
+- In Databricks, SparkR benefits from the custom enhancements made to Spark in the Databricks Runtime.
+- Lacks full support of Spark MLib - in particular missing the feature engineering functions. 
+
+### sparklyr
+- Good support for data science use cases; full set of feature engineering and distributed ML algorithms. 
+- Compatibility with dplyr syntax reduces learning curve and makes it easy to switch between writing distributed queries on Spark and vanilla queries on R dataframes.
+- Lacks full support of Spark APIs without extension. For example, SparkR's `to_json()` has no equivalent in sparklyr, whilst dealing with nested data requires the package `sparklyr.nested`.
+
+In summary, choosing whether to use sparklyr or SparkR should depend on what you are trying to do and may call for flexibility even within the same project. Both projects are open source allowing users to inspect and contribute to the code base, both have good documentation and user communities for seeking help.
+
+## Is there a performance difference between sparklyr and SparkR?
```

Comment:
On Databricks there is an out of the box performance optimization when transferring data between SparkR and R.  This works when collecting data, pushing data to Spark, or running UDFs. 

With sparklyr you can get the same benefits by installing Apache Arrow (link to install guide).

Anecdotally, we have observed some performance issues when using windowing functions with sparklyr compared to SparkR.  We've also observed that UDFs are more scalable and stable with SparkR.

Reply:
Thanks. I'll have a look at running some tests to quantify those points and see how much of a difference it makes

Reply:
It's basically the difference between having Arrow on and not (so, orders of magnitude).  Like 40 seconds vs. 40 minutes.